grub-core/loader/multiboot_mbi2.c

```diff
@@ -1214,16 +1227,19 @@ grub_multiboot2_prepare_slaunch_txt (grub_uint32_t mbi_target,
                                       mbi_size,
                                       "Measured MB2 information");
   grub_slaunch_add_slrt_policy_entries ();
-  grub_txt_add_slrt_policy_entries ();
+  if (slp == SLP_INTEL_TXT)
+    grub_txt_add_slrt_policy_entries ();
   add_multiboot2_slrt_policy_entries ();
   grub_slaunch_finish_slr_table ();
 
   grub_dprintf ("multiboot_loader", "slr_table_base = %lx, slr_table_size = %x\n",
                 (unsigned long) slparams->slr_table_base,
                 (unsigned) slparams->slr_table_size);
-  grub_memcpy ((void *)(grub_addr_t) slparams->slr_table_base,
-               slparams->slr_table_mem,
-               slparams->slr_table_size);
+
+  if (slp == SLP_INTEL_TXT)
```

Comment:
This should happen for AMD also.

Reply:
Why? `grub_skinit_boot_prepare()` assigns `slparams->slr_table_base` to physical and `slparams->slr_table_mem` to corresponding virtual address, there is no intermediate buffer for SLRT in this case.

Reply:
You are correct. Not sure why I had to do this in testing, possibly `grub-install` failed between one version and another and I didn't notice.